File: 026-remove-duplicates-from-sorted-array.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/

class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def removeDuplicates(self, nums):

        i = 0
        u = 0

        while i < len(nums):
            if i != u:
                nums[u] = nums[i]

            while i < len(nums) - 1 and nums[i] == nums[i + 1]:
                i = i + 1

            u = u + 1
            i = i + 1

        return u

File: 026-remove-duplicates-from-sorted-array.js
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
	var i = 0, u = 0;

	while (i < nums.length) {

		if (i !== u) {
			// equal to: nums[u] = nums[i];
			nums.splice(u, 1, nums[i]);
		}

		while (i < nums.length - 1 && nums[i] === nums[i + 1]) {
			i++;
		}

		u++;
		i++;
	}

	return u;
};
